cli/azd/pkg/devcenter/environment_store.go

```diff
@@ -174,7 +177,19 @@ func (s *EnvironmentStore) Reload(ctx context.Context, env *environment.Environm
 // DevCenter doesn't implement any APIs for saving environment configuration / metadata
 // outside of the environment definition itself or the ARM deployment outputs
 func (s *EnvironmentStore) Save(ctx context.Context, env *environment.Environment) error {
-	return nil
+	if s.config.Project != "" {
+		if err := env.Config.Set(DevCenterProjectPath, s.config.Project); err != nil {
+			return err
+		}
+	}
+
+	if s.config.EnvironmentType != "" {
+		if err := env.Config.Set(DevCenterEnvTypePath, s.config.EnvironmentType); err != nil {
+			return err
+		}
+	}
+
+	return s.local.Save(ctx, env)
```

Comment:
Allows local environment configuration to update the project/env type if not already set.